### ISSUE TITLE
Invalid Template Deployment

### ISSUE THREAD

**Issue by prakhargurawa** [Azure-Samples/azure-search-openai-demo#649]
# InvalidTemplateDeployment

### Hi I just performed all the step required which are:

- Create a new folder and switch to it in the terminal
- azd auth login
- azd init -t azure-search-openai-demo
- In the data folder i changed to my documents (PDF).
- In the file infra/main.parameters I changed from AZURE_SEARCH_SERVICE_SKU=standard to AZURE_SEARCH_SERVICE_SKU=basic as i need basic version of cognitive search.
- Finally I am running azd up and it is throwing **InvalidTemplateDeployement**.

<img width="989" alt="CaptureInvalid" src="https://github.com/Azure-Samples/azure-search-openai-demo/assets/10551751/02b2f992-b25e-4c3c-b355-b01fc607c789">

### This issue is for a: (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> I just did all the basic steps which were on repo.

### Any log messages given by the failure
> Attached screenshot already, but still the text message is:

ERROR: deployment failed: failing invoking action 'provision', error deploying infrastructure: deploying to subscription:

```
Deployment Error Details:
InvalidTemplateDeployment: The template deployment 'openai' is not valid according to the validation procedure. The tracking id is 'ff48d46b-13d3-4d1d-969b-52da9abbd296'. See inner errors for details.
FlagMustBeSetForRestore: An existing resource with ID '/subscriptions/19866297-20c6-4921-b541-e867aeda8d3f/resourceGroups/rg-Tenders-GPT-Bot-dev/providers/Microsoft.CognitiveServices/accounts/cog-mt66akh7fgwb4' has been soft-deleted. To restore the resource, you must specify 'restore' to be 'true' in the property. If you don't want to restore existing resource, please purge it first.
InvalidTemplateDeployment: The template deployment 'formrecognizer' is not valid according to the validation procedure. The tracking id is 'cc971759-8174-4151-ba43-ddd6b865c623'. See inner errors for details.
FlagMustBeSetForRestore: An existing resource with ID '/subscriptions/19866297-20c6-4921-b541-e867aeda8d3f/resourceGroups/rg-Tenders-GPT-Bot-dev/providers/Microsoft.CognitiveServices/accounts/cog-fr-mt66akh7fgwb4' has been soft-deleted. To restore the resource, you must specify 'restore' to be 'true' in the property. If you don't want to restore existing resource, please purge it first.
```

### Expected/desired behavior
> I have used this repo thrice for creating 3 different bots and it worked everytime before.

### OS and Version?
> Windows 10

### azd version?
azd version 1.1.0

### Mention any other details that might be useful
- In the file infra/main.parameters I changed from AZURE_SEARCH_SERVICE_SKU=standard to AZURE_SEARCH_SERVICE_SKU=basic as i need basic version of cognitive search.
> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by pamelafox** [comment]
Did you previously deploy this template and take it down? Can you go into Azure Portal and look to see if you have any soft-deleted Cognitive Services accounts that need purging? That's what the error message seems to indicate.

**Comment by pamelafox** [comment]
Thanks for filing this and sorry you ran into deployment issues.

From the error details, this appears to be environment-specific rather than a repo bug:
- FlagMustBeSetForRestore indicates there are soft-deleted Cognitive Services accounts with the same names in your subscription/resource group. Azure blocks creating a new account with the same name until the deleted one is purged or restored.

To unblock:
- In Azure Portal, check Cognitive Services Deleted accounts (or the resource group) and purge the deleted accounts listed in the error (e.g., cog-mt66akh7fgwb4, cog-fr-mt66akh7fgwb4). Alternatively, create a new azd environment so unique resource names are generated (e.g., `azd env new myenv2` then `azd up`).
- Make sure you're on a recent `azd` version.
- Note: Basic SKU for Azure AI Search typically does not support vector search features used by this sample; Standard S1 or higher is recommended.

We haven't heard back since the original guidance and the issue has been inactive for some time, so we're closing this to keep the tracker tidy. If you're still hitting this after purging/restoring soft-deleted resources and using the latest template, please comment with updated details or open a new issue and we'll be happy to help.
